### PR TITLE
move Sub&Pub master from MultiCameraState to camera_common

### PR DESCRIPTION
**Pull request by deanlee**
This is part of https://github.com/commaai/openpilot/pull/20137 , split for easier review

### CODE REVIEW

File: selfdrive/camerad/cameras/camera_common.cc [hunk 1]
```diff
@@ -348,16 +353,17 @@ std::thread start_process_thread(MultiCameraState *cameras, CameraState *cs, pro
...
   const CameraBuf *b = &c->buf;
 
   static int x_min = 0, x_max = 0, y_min = 0, y_max = 0;
   static const bool is_rhd = Params().read_db_bool("IsRHD");
 
   // auto exposure
   if (cnt % 3 == 0) {
-    if (sm->update(0) > 0 && sm->updated("driverState")) {
-      auto state = (*sm)["driverState"].getDriverState();
+    if (sm.updated("driverState")) {
```
Comment (pd0wm) [inline selfdrive/camerad/cameras/camera_common.cc:365]: What calls sm.update() now?

Comment (deanlee) [reply selfdrive/camerad/cameras/camera_common.cc:365]: my bad

File: selfdrive/camerad/cameras/camera_webcam.cc [hunk 2]
```diff
@@ -243,25 +242,24 @@ void cameras_open(MultiCameraState *s) {
 void cameras_close(MultiCameraState *s) {
   camera_close(&s->road_cam);
   camera_close(&s->driver_cam);
-  delete s->pm;
 }
 
-void process_driver_camera(MultiCameraState *s, CameraState *c, int cnt) {
+void process_driver_camera(MultiCameraState *s, CameraState *c, PubMaster *pm, int cnt) {
```
Comment (adeebshihadeh) [inline selfdrive/camerad/cameras/camera_webcam.cc:247]: pass by const ref?

**Comment by adeebshihadeh** [comment]
Can you rebase?

**Comment by deanlee** [comment]
done

File: selfdrive/camerad/cameras/camera_common.cc [hunk 3]
```diff
@@ -318,6 +318,11 @@ float set_exposure_target(const CameraBuf *b, int x_start, int x_end, int x_skip
 extern ExitHandler do_exit;
 
 void *processing_thread(MultiCameraState *cameras, CameraState *cs, process_thread_cb callback) {
+  static PubMaster pm({"roadCameraState", "driverCameraState", "thumbnail"
```
Comment (adeebshihadeh) [inline selfdrive/camerad/cameras/camera_common.cc:321]: why is this static?

Comment (deanlee) [reply selfdrive/camerad/cameras/camera_common.cc:321]: processing_thread will be called from multiple camera threads.use static to make sure Pubmaster only initialized once

**Pull request closed without merging**